src/main/java/io/appform/dropwizard/actors/compression/CompressionProvider.java

```diff
@@ -0,0 +1,54 @@
+package io.appform.dropwizard.actors.compression;
+
+import org.apache.commons.jcs.utils.zip.CompressionUtil;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPOutputStream;
+
+public class CompressionProvider {
+
+    public static byte[] compress(byte[] data, CompressionAlgorithm compressionAlgorithm) throws IOException {
+
+        return compressionAlgorithm.visit(new CompressionAlgorithmVisitor<byte[]>() {
+
+            @Override
+            public byte[] visitGzip() throws IOException {
+                return compressUsingGzip(data);
+            }
+
+            @Override
+            public byte[] visitDeflate() throws IOException {
+                return CompressionUtil.compressByteArray(data);
+            }
+        });
+    }
+
+
+    public static byte[] decompress(byte[] data, CompressionAlgorithm compressionAlgorithm) throws IOException {
```

Comment:
Why static? Refactor the code as mentioned above to implement a decorator/transformer type construct to avoid these.

Also.. add tests.